cmd/noms/diff/diff.go

```diff
@@ -23,7 +23,7 @@ func shouldDescend(v1, v2 types.Value) bool {
 }
 
 func Diff(w io.Writer, v1, v2 types.Value) error {
-	return diff(w, types.NewPath(), nil, v1, v2)
+	return diff(w, types.Path{}, nil, v1, v2)
```

Comment:
I want to think of Paths now as more literally []PathPart, with some helper functions. So removing NewPath and just using Path{} is more straightforward.